kedro/framework/cli/starters.py

```diff
@@ -849,7 +849,7 @@ def _create_project(template_path: str, cookiecutter_args: dict[str, Any]):
 
     # we can use starters without tools:
     if tools is not None:
-        if tools in ("[]", ""):  # TODO: This should be a list
+        if tools in ("['None']", "[]"):  # TODO: This should be a list
```

Comment:
Where is this ["None"] come from? At the very end of the process I expected all the tool should be converted to a list of names that is ready to be copy to `pyproject.toml`.

What's the expected result of for the following cases? 
1. `kedro new --tools=none
2. `kedro new` -> Prompt enter
```
# pyproject.toml
tools = 
```

Reply:
Looks that right now it will be "[]" after Prompt and ['None'] after CLI. We should unify that

Reply:
It comes from starters.py L521;

Previously, when storing the selected tools as a string, when the input "none" was parsed through CLI or config, an empty list wrapped in a string would be returned `"[]"` , and if the selection for no tools was made through the prompts, an empty string `""` would be returned instead. 

We now store the tools selection by passing through to cookiecutter a list wrapped in a string, that we then unwrap in the template `pyproject.toml`. This resulted in errors as unwrapping an empty string (from no tools selection) would result in an invalid toml file.

The implementation results in two different values depending on how the selection for no tools is made, after #/3426 the values are as such:
1. "None" is provided through the flag --tools, or config -> the result of parsing this selection will end in L521 in starters.py storing tools as str(['None'])
2. Selecting "none" through the interactive flow -> the prompt will be parsed by `_parse_tools_input` instead, and will be registered as an empty list `[]`, which is then wrapped in a string and stored as "[ ]"

As such, after #/3426, the condition has changed from `("[]", "")` to `("['None']", "[]")`

Undoubtedly this code requires some refactoring 😬; I am happy to make changes to `_parse_tools_input` within this PR, but an overarching refactor (or code evaluation) is still necessary

Reply:
Thank you, @AhdraMeraliQB . The current situation with the tool's processing is really unclear. I agree it's a good idea to add that refactoring task to minor tech debts ticket.